### DISCUSSION TITLE
Manual stats updates

### DISCUSSION THREAD

**Discussion by Sboshoff76** [celsworth/lxp-bridge#197] [Ideas]
The default is currently about 5 minutes. The Lux website has this refresh button where you can force an update which shows the latest values on the website (and in HA).
![image](https://github.com/celsworth/lxp-bridge/assets/68126343/e06dd840-8832-4935-afb7-b75a0fafb315)

Is it perhaps possible for you to call this function from within HA using your integration? It could then be put onto a minute or 30 second automation schedule to provide more real-time data.

**Comment by jacovt** [comment]
I am currently driving updates from HA on a 5 second interval by sending a payload of "1" to lxp/cmd/BAxxxxxxx/read/inputs/{1,2,3} which basically triggers the updates to be published out to MQTT. 

Also see this: https://github.com/celsworth/lxp-bridge/wiki/MQTT-Publishes

This gives me near-realtime views in HA of what is happening on the inverters.

Have you tried this yet?

**Reply by Sboshoff76** [reply]
Right at the bottom of the link you gave me is this bit:
![image](https://github.com/celsworth/lxp-bridge/assets/68126343/689d7f6b-ff0b-4f45-913f-eaa2c7ff810b)

@celsworth does that mean if you put parameter 0 as 20, it'll update the stats every 20 seconds and you don't need to send payloads liek Jaco is doing at all ?

**Reply by celsworth** [reply]
In theory, yes :) I've previously heard from one person who has set this parameter and gotten it working in this way.

Only thing to be aware of is that Luxpower will see your frequent stats updates as well, so they'll end up logging a data row in their web portal every 20 seconds for you. Which potentially starts filling their storage up if enough people do it (clearly one or two people won't make a lot of difference). And if it annoys them enough remember they have complete remote control over your entire inverter.

I know a few people have started isolating their inverters from the Internet to avoid all these problems (you'll just need to turn on the local heartbeats option in lxp-bridge config then).

**Reply by Sboshoff76** [reply]
Is this param (or even jaco's implementation) something you'll consider adding perhaps? If not does anyone have some advice on how to set this. I was looking yesterday for some guides but it was all a foreign language to me.

**Reply by celsworth** [reply]
I'm not too sure about exposing this to HA. I'd suggest manually pushing an MQTT payload to set it. You can do this from a commandline with something like:

```
mosquitto_pub -h your.mqtt.host.ip -t lxp/cmd/all/set/param/0 -m 20
```

The `all` here is usually an inverter datalog serial number, but the magic "all" is easier and will send the command to all inverters lxp-bridge is connected to (probably just one, unless you're doing parallel stuff).

Alternatively, http://mqtt-explorer.com is a nice easy to use graphical interface into MQTT. The important parts to know are the topic (`lxp/cmd/all/set/param/0`) and the payload to send (`20`).

**Reply by EddieSpaghettie** [reply]
Sorry how exactly do you do this?

**Comment by Sboshoff76** [comment]
Ahh.. I haven't played around with pushing payloads yet. Thanks for the info!

**Comment by chamsters** [comment]
In case anyone is reaching this topic a while later, the answer is in the thread above :)

1) publish: mosquitto_pub -h your.mqtt.host.ip -t lxp/cmd/all/set/param/0 -m <<TIME IN SECONDS>> to set the refresh interval
2) I've seen that the data logger on the lux side is still only adding a record every 5 minutes, although I'm seeing MQTT published every xx seconds as defined above
3) My data logger was previously only refreshing every 5 minutes. No issues at 30 seconds now!

**Comment by andrewfraley** [comment]
To save anyone else the trouble, this is how you would run this

```
sudo apt install mosquitto-clients
mosquitto_pub -h homeassistant.localdomain -t lxp/cmd/all/set/param/0 -m 20 -u mqtt -P somepassword
```

Also I confirmed that this works with an EG4 12000XP and I'm now getting data every 20 seconds instead of every 120 seconds, nice!
